Heisenberg4269/Trie.cpp

```diff
@@ -0,0 +1,59 @@
+class Trie
+{
+public:
+    struct tnode
+    {
+        vector <tnode*> child;
+        bool isword;
+    };
+    tnode *root;
+    int vocabLength;
+
+    Trie(int vocab = 26)
+    {
+        vocabLength = vocab;
+        root = makeTnode();
+    }
+
+    tnode* makeTnode()
+    {
+        tnode *newTnode = new tnode;
+        newTnode->child.resize(vocabLength);
+        for(int i = 0; i<vocabLength; i++)
+        {
+            newTnode->child[i] = NULL;
+        }
+        newTnode->isword = false;
+        return newTnode;
+    }
+
+    void insert(string str)
+    {
+        tnode *temp = root;
+        for(int i = 0; i<str.size(); i++)
```

Comment:
Try using a variable instead of `str.size()` if the size remains constant. `.size()` is a member function of string class that gets called again

Reply:
Also, avoid using `temp` for variable names. It is a bad practice.